app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,18 @@
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  include OrganizationScopable
+
+  def google_oauth2
+    @user = User.from_omniauth(request.env["omniauth.auth"])
+    if @user.persisted?
+      sign_in_and_redirect @user, event: :authentication
+      set_flash_message(:notice, :success, kind: "Google") if is_navigational_format?
+    else
+      session["devise.google_data"] = request.env["omniauth.auth"].except(:extra)
+      redirect_to new_user_registration_url, alert: @user.errors.full_messages.join("\n")
+    end
+  end
+
+  def failure
```

Comment:
Where does this action get used? There's no route for it and I don't see it referenced in the action above it.